src/resonanceReconstruction/breitWigner/CrossSection.hpp

```diff
@@ -26,5 +26,19 @@ struct CrossSection {
              this->capture - other.capture,
              this->fission - other.fission };
   }
+
+  template< typename T >
+  CrossSection operator*( const T& other ) const {
```

Comment:
This is used in the linearization when calculating the `trial`

Reply:
Any type other than a unitless quantity or a floating point value or integer value will fail to compile since the constructor of CrossSection is expecting a quantity of barns. I assume that you are aware of this?

It begs the question if this should actually be in here, or if it should be defined in RECONR itself.

Reply:
Yeah, I learned that doing this for non-unitless quantity failed to compile. I could do this in RECONR, but thought that since it was already here, I could just add it here.

Reply:
This is also needed for use with [twig](https://github.com/njoy/twig). In there it [has](https://github.com/njoy/twig/blob/96e69878118280c6d8d00dcc0809cbfc5dcb9e4e/src/twig/linearize/Callable/src/bin.hpp#L12)
```cpp
const auto trial = 0.5 * ( y[left] + y[right] );
```
where `y[left]` and `y[right]` are `CrossSection` objects. In order to multiple by `0.5` This functionality needs to be in there. We could write it such that it takes a double instead of a type `T`.

Reply:
I know why it is needed :-) 
Writing code using a template that will fail in 99.9% of all types that you can use it with is asking for trouble. Had you embedded this into RECONR, I would have accepted it without hesitation.
Using a double is probably more prudent, or use an enable_if for a type T for which T * Quantity<barn> = Quantity<barn>. I don't know if this is something you can specify.

Now, we could modify twig to accept a function/functor to calculate the average in the same way as the tolerance, and use a lambda that calculates the default average as 0.5*(left+right).